tests/Http/GroupTypeTest.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace Tests\Http;
+
+use Tests\TestCase;
+use Rogue\Models\GroupType;
+use Illuminate\Database\QueryException;
+
+class GroupTypeTest extends TestCase
+{
+    /**
+     * Test that a GET request to /api/v3/group-types returns an index of all group types.
+     *
+     * @return void
+     */
+    public function testGroupTypeIndex()
+    {
+        $groupTypes = factory(GroupType::class, 5)->create();
+
+        $response = $this->getJson('api/v3/group-types');
+        $decodedResponse = $response->decodeResponseJson();
+
+        $response->assertStatus(200);
+        $this->assertEquals(5, $decodedResponse['meta']['pagination']['count']);
+
+        foreach ($groupTypes as $groupType) {
+            $this->assertDatabaseHas('group_types', [
+                'name' => $groupType->name,
+            ]);
+        }
+    }
+
+    /**
+     * Test that the group_type name field is unique.
+     *
+     * @return void
+     */
+    public function testUniqueGroupTypeNameIndex()
+    {
+        $this->expectException(QueryException::class);
+
+        $groupType = factory(GroupType::class)->create();
+
+        factory(GroupType::class)->create([
+            'name' => $groupType->name,
+        ]);
```

Comment:
Similarly, I think we could safely omit this test & instead test that the endpoint or command we'd be using to create these records returns a good error (e.g. if there was a "create" endpoint, we might check `$response->assertStatus(422)`).

Speaking of, how are we planning on creating new group types when this is in prod?

Reply:
Yeah, I figured a `POST / request` of some sort would be a better test. I'm thinking only allowing admins, and also was thinking about modeling how actions and campaigns are currently created via server-side blade templates, vs mutations. mainly for sake of convenience. we'll have the same issue when its time to create `groups` once that model is in place. Do you have a strong preference for all GraphQL? The server-side approach I'm thinking would be even more TODO's, but feels like it'd go faster... maybe I'm wrong though.

Reply:
My preference would definitely be to use GraphQL mutations, since it better sets us up for extracting this admin interface in the future (and we'll probably need GraphQL stuff anyways for the user-facing views).

That said, I can totally understand that it's faster to throw together some of these CRUD-y UIs in this one single repository rather than jumping around, so that's a fair trade-off if you want to make it! Just will add a lil' bit to that epic down the line if we can ever get to it. 🙏

One alternate idea to keep things quick & simple – would an Artisan command work for adding these (like `php artisan rogue:group-type:add` or something?)

Reply:
The artisan command is a great idea --thanks! I liked the idea of adding the web controller to have an example for when we create/edit groups next sprint, but I think it'll be easy enough to reference the actions or campaigns controllers.